packages/core/src/lib/map/map.ts

```diff
@@ -130,18 +130,19 @@ export function mapReturn<
   errorHandler: ErrorHandler,
   isMapArray = false
 ): TDestination {
-  return map(
+  return map({
     sourceObj,
     mapping,
     options,
     mapper,
     errorHandler,
-    (destinationMemberPath: string[], destination: TDestination) =>
-      (value: unknown) => {
-        destination = set(destination, destinationMemberPath, value);
-      },
-    isMapArray
-  );
+    setMemberFn:
```

Comment:
suggestion: let's also reuse this one if possible

Reply:
Can you explain more detail about this reuse?

Reply:
It is same code here, just indent change, did you mean to extract L:142 and L:367 to function?

Reply:
Yes. extract L:140 to a function and reuse it. Like what you did for `mapMutate()`

Reply:
It take a little time to understand why to assign `destination` variable.
Usually, parameter assign may be a bad smell.
```
destination = set(destination, destinationMemberPath, value);
```